### ISSUE TITLE
Adding quotes to args in docker-compose causes the values to be ignored

### ISSUE THREAD

**Issue by ahasna** [ethereum/go-ethereum#26411]
### Background

I am building a dockerized application to deploy private networks using `geth`. For that I am using `docker-compose` file, and the idea is to run `docker-compose up` once the machine is deployed which will create a plug-and-play private ethereum network for some internal testing for contracts deployments. 

### The initial docker-compose file I used

```yaml
version: "3.8"

services:
  geth:
    image: ethereum/client-go
    pull_policy: if_not_present
    container_name: geth
    restart: unless-stopped
    ports:
      - 30303:30303
      - 30303:30303/udp
      - 8545:8545
      - 8546:8546
      - 8551:8551
    stop_signal: SIGINT
    stop_grace_period: 2m
    healthcheck:
      test: ["CMD-SHELL", "echo hi"]
      interval: 10s
      timeout: 5s
      retries: 5
    volumes:
      - ./geth/geth_password:/geth/password
    command:
      - --datadir=/geth/data
      - --networkid=12345
      - --allow-insecure-unlock
      - --unlock=0x000...
      - --password=/geth/password
      - --mine
      - --miner.threads=1
      - --nodiscover
      - --gcmode=archive
      - --http
      - --http.addr=0.0.0.0
      - --http.vhosts=*
      - --http.corsdomain=*
      - --http.api="web3,eth,debug"
      - --rpc.allow-unprotected-txs
      - --nodiscover
      - --ws
      - --ws.addr=0.0.0.0
      - --ws.origins=*
      - --ws.api="web3,eth,debug"
```

### The problem

Starting this container was fine and I could see blocks being mined. However when I tried to deploy contracts I got this error message:

```log
the method web3_clientVersion does not exist/is not available
```

After a tedious two-day debugging , I learned (by randomly guessing) that the quotes in the following two cli arguments where causing `geth` to ignore them or maybe not parsing them correctly:

```txt
      - --http.api="web3,eth,debug"
      - --ws.api="web3,eth,debug"
```
### The solution
Removing the quotes fixed the issue and I could deploy the contracts successfully.

#### Working Docker File
```yaml
version: "3.8"

services:
  geth:
    image: ethereum/client-go
    pull_policy: if_not_present
    container_name: geth
    restart: unless-stopped
    ports:
      - 30303:30303
      - 30303:30303/udp
      - 8545:8545
      - 8546:8546
      - 8551:8551
    stop_signal: SIGINT
    stop_grace_period: 2m
    healthcheck:
      test: ["CMD-SHELL", "echo hi"]
      interval: 10s
      timeout: 5s
      retries: 5
    volumes:
      - ./geth/geth_password:/geth/password
    command:
      - --datadir=/geth/data
      - --networkid=12345
      - --allow-insecure-unlock
      - --unlock=0x000...
      - --password=/geth/password
      - --mine
      - --miner.threads=1
      - --nodiscover
      - --gcmode=archive
      - --http
      - --http.addr=0.0.0.0
      - --http.vhosts=*
      - --http.corsdomain=*
      - --http.api=web3,eth,debug
      - --rpc.allow-unprotected-txs
      - --nodiscover
      - --ws
      - --ws.addr=0.0.0.0
      - --ws.origins=*
      - --ws.api=web3,eth,debug
```

### My question

Is it actually the case that quotes were causing the issue?

**Comment by karalabe** [comment]
The quotes were passed verbatim to Geth, not stripped by docker, so we interpreted it as 3 modules: `"web3`, `eth` and `debug"`. Please make sure that the quotes don't end up in the parameter value, we don't want to sanitize it inside Geth since it's a bad call in the first place.

Also, the reason we did comma separated values instead of space separated ones was specifically to not require quotes, which are always weird when passed through multiple shells / systems.

Tl;DR: Yes, the quotes are bad. Just don't use them and it will be fine.
